### PR TITLE
Improve full-stack.t

### PR DESCRIPTION
**Pull request by okurz**
* t: Simplify OpenQA::Test::FullstackUtils
* t: Use signatures in OpenQA::Test::FullstackUtils
* t: Add timeout to developer console waiting status log message

### CODE REVIEW

File: t/lib/OpenQA/Test/FullstackUtils.pm [hunk 1]
```diff
@@ -44,52 +43,42 @@ my $JOB_SETUP = {
...
+sub wait_for_result_panel (
+    $driver, $result_panel,
+    $context = 'current job',
+    $fail_on_incomplete = undef,
+    $check_interval = 0.5
+  )
+{
     my $looking_for_result = $result_panel =~ qr/Result: /;
-    $context //= 'current job';
     my $msg = "Expected result for $context not found";
```
Comment (okurz) [inline t/lib/OpenQA/Test/FullstackUtils.pm:81]: CI tests show a warning about `$context` not defined, see https://app.circleci.com/pipelines/github/os-autoinst/openQA/8499/workflows/0d301351-4632-4b07-bf9f-e2036ff8c24b/jobs/80265?invite=true#step-110-99 , but https://github.com/os-autoinst/openQA/pull/4379/files#diff-523eaf01cc7dd1b0371426b1d310936bc8ae615a6b15cbd107ea0364ccbf8bdaR75 shows that there should be a default value. Did I do it wrong?

Comment (Martchus) [reply t/lib/OpenQA/Test/FullstackUtils.pm:81]: Likely `undef` (or a variable with `undef` value) is passed explicitly by some invocation. Adding a signature with default value is *not* equivalent to assigning a "default" via `//=`.

Comment (okurz) [reply t/lib/OpenQA/Test/FullstackUtils.pm:82]: ok, using `undef` as default and setting the string accordingly.

**Review by Martchus — APPROVED** [review]
Should be good, only tidy doesn't like it.

**Review by kalikiana — APPROVED** [review]

**Pull request merged**